source/proof_document/lib/proof_document.rb

```diff
@@ -0,0 +1,156 @@
+require 'bundler/setup' if !defined?(Bundler)
+require 'faraday'
+require 'json'
+require 'retries'
+require '/opt/ruby/lib/ssm_helper' if !defined?(IdentityIdpFunctions::SsmHelper)
+require 'openssl'
+require "aws-sdk-s3"
+
+module IdentityIdpFunctions
+  class ProofDocument
+    include IdentityIdpFunctions::FaradayHelper
+
+    def self.handle(event:, context:, &callback_block)
+      params = JSON.parse(event.to_json, symbolize_names: true)
+      new(**params).proof(&callback_block)
+    end
+
+    attr_reader :encryption_key, :front_image_iv, :back_image_iv, :selfie_image_iv,
+                :front_image_url, :back_image_url, :selfie_image_url,
+                :liveness_checking_enabled, :callback_url
+
+    def initialize(encryption_key:,
+                   front_image_iv:,
+                   back_image_iv:,
+                   selfie_image_iv:,
+                   front_image_url:,
+                   back_image_url:,
+                   selfie_image_url:,
+                   liveness_checking_enabled:,
+                   callback_url:)
+      @encryption_key = encryption_key
+      @front_image_iv = front_image_iv
+      @back_image_iv = back_image_iv
+      @selfie_image_iv = selfie_image_iv
+      @front_image_url = front_image_url
+      @back_image_url = back_image_url
+      @selfie_image_url = selfie_image_url
+      @liveness_checking_enabled = liveness_checking_enabled
+      @callback_url = callback_url
+    end
+
+    def proof(&callback_block)
+      set_up_env!
+
+      proofer_result = with_retries(**retry_options) do
+        document_proofer.post_images(
+          front_image: read_file(front_image_url, front_image_iv),
+          back_image: read_file(back_image_url, back_image_iv),
+          selfie_image: read_file(selfie_image_url, selfie_image_iv),
```

Comment:
1. do we need to make this read from s3 conditional?

Reply:
Can you be more specific?  Do you mean wrt if a read_file_fails?

Reply:
I mean, should we check that `selfie_image_url` exists before we try to read from it?

At least for the time being selfie is still optional, right? So I'm thinking something like:

```
selfie_image: liveness_checking_enabled? && read_file(selfie_image_url, selfie_image_iv)
```

Reply:
Nice.  Good catch.